tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerGroupExecutor.java

```diff
@@ -0,0 +1,211 @@
+                                                           Optional<Properties> customConfigs,
+                                                           boolean syncCommit) {
+        return new ConsumerGroupExecutor(
+                brokerAddress,
+                numConsumers,
+                groupId,
+                groupProtocol,
+                topic,
+                RangeAssignor.class.getName(),
+                remoteAssignor,
+                customConfigs,
+                syncCommit
+        );
+    }
+
+    public static ConsumerGroupExecutor buildClassicGroup(String brokerAddress,
+                                                          int numConsumers,
+                                                          String groupId,
+                                                          String topic,
+                                                          String assignmentStrategy,
+                                                          Optional<Properties> customConfigs,
+                                                          boolean syncCommit) {
+        return new ConsumerGroupExecutor(
+                brokerAddress,
+                numConsumers,
+                groupId,
+                GroupProtocol.CLASSIC.name,
+                topic,
+                assignmentStrategy,
+                Optional.empty(),
+                customConfigs,
+                syncCommit
+        );
+    }
+
+    void submit(ConsumerRunnable consumerThread) {
+        consumers.add(consumerThread);
+        executor.execute(consumerThread);
+    }
+
+    @Override
+    public void close() throws Exception {
+        consumers.forEach(ConsumerRunnable::shutdown);
+        executor.shutdown();
+        try {
+            executor.awaitTermination(5000, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static class ConsumerRunnable implements Runnable {
+        private final String brokerAddress;
+        private final String groupId;
+        private final Properties customConfigs;
+        private final boolean syncCommit;
+        private final String topic;
+        private final String groupProtocol;
+        private final String assignmentStrategy;
+        private final Optional<String> remoteAssignor;
+        private final Properties props = new Properties();
+        private KafkaConsumer<String, String> consumer;
+        private boolean configured = false;
+        private volatile boolean isShutdown = false;
+
+        public ConsumerRunnable(String brokerAddress,
+                                String groupId,
+                                String groupProtocol,
+                                String topic,
+                                String assignmentStrategy,
+                                Optional<String> remoteAssignor,
+                                Optional<Properties> customConfigs,
```

Comment:
`Map<String, Object>` is more suitable in this case. The `empty map` can be equal to `Optional.empty` but it gets more graceful.